File: experimental/workflow_mt/producer/typescript/src/producer/beacons/structa.ts
import {
    Context,
    Producer,
    Identification,
    Filter,
    Protocol,
} from "../implementation/beacons";

export function emit(
    beacon: Protocol.StructA,
    consumer: Identification,
    filter: Filter,
    context: Context,
    producer: Producer
): Promise<void> {
    return Promise.reject(
        new Error(`Handler for Protocol.StructA isn't implemented.`)
    );
}

File: tests/workflow/consumer/rust/src/consumer/broadcasts/structa.rs
use super::{protocol, Consumer, Context};
use clibri::client;

pub async fn handler<E: client::Error>(
    event: protocol::StructA,
    context: &mut Context,
    consumer: Consumer<E>,
) {
    context.broadcast.structa += 1;
    context.broadcast.check();
}